channels/factories.py

```diff
@@ -39,3 +160,131 @@ class Params:
                 lambda: FAKE.date_time_this_year(before_now=True, after_now=False, tzinfo=pytz.utc)
             )
         )
+
+
+class ChannelFactory(factory.Factory):
+    """Factory for channels"""
+    api = None
+    title = factory.Faker('text', max_nb_chars=50)
+    public_description = factory.Faker('text', max_nb_chars=100)
+    channel_type = FuzzyChoice([CHANNEL_TYPE_PUBLIC, CHANNEL_TYPE_PRIVATE])
+
+    @factory.lazy_attribute
+    def name(self):
+        """Lazily determine a unique channel name"""
+        now = now_in_utc().timestamp()
+        return "{}_{}".format(int(now), FAKE.word())[:21]  # maximum of 21-char channel names
+
+    @factory.post_generation
+    def _api(self, *args, **kwargs):  # pylint: disable=unused-argument
+        """Lazily create the channel"""
+        assert self.api, "ChannelFactory requires an api instance"
+
+        self.api.create_channel(
+            self.name,
+            self.title,
+            channel_type=self.channel_type,
+            public_description=self.public_description
+        )
+
+    class Meta:
+        model = Channel
+
+
+class PostFactory(factory.Factory):
+    """Abstract factory for posts"""
+    api = None
+    title = factory.Faker('text', max_nb_chars=50)
+
+    @factory.lazy_attribute
+    def channel(self):
+        """Lazily create a channel"""
+        assert self.api, "PostFactory requires an api instance"
+
+        return ChannelFactory.create(api=self.api)
+
+    class Meta:
+        abstract = True
+
+
+class TextPostFactory(PostFactory):
+    """Factory for text posts"""
+    text = factory.Faker('text', max_nb_chars=100)
+
+    @factory.lazy_attribute
+    def id(self):
```

Comment:
Because the model is a `namedtuple`, which is an immutable type and we need to get the id from the reddit backend. An alternative would be to write out a whole class implementation for each object type, but that seemed a tad heavyweight for what we need here.

Reply:
Fine with me